crates/services/producer/src/block_producer.rs

```diff
@@ -352,8 +352,10 @@ where
             .consensus_parameters_provider
             .consensus_params_at_version(&block_header.consensus_parameters_version)?
             .block_gas_limit();
+        // We have a hard limit of u16::MAX transactions per block, including the final mint transactions.
+        // Therefore we choose the `new_da_height` to never include more than u16::MAX - 1 transactions in a block.
         let new_da_height = self
-            .select_new_da_height(gas_limit, previous_da_height)
+            .select_new_da_height(gas_limit, previous_da_height, u16::MAX - 1)
```

Comment:
maybe this should be `u16::MAX`? We can have `65_536` tx in a block (`u16::MAX + 1`), hence without the mint transaction this becomes `u16::MAX`.

Reply:
This seems fine.